main.py

```diff
@@ -34,12 +34,7 @@ def main():
         return
 
     if args.verbose:
-        print(f"INITIAL NODES:")
```

Comment:
would it be possible to leave the original --verbose behavior as-is, and add command line flags --visualize_initial <filename> and --visualize-modified <filename> that cause the visualization to be generated?